client/blocks/upwork-stats-nudge/index.js

```diff
@@ -0,0 +1,140 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal Dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import isUpworkStatsNudgeDismissed from 'state/selectors/is-upwork-stats-nudge-dismissed';
+import QueryPreferences from 'components/data/query-preferences';
+import SectionHeader from 'components/section-header';
+import { dismissNudge } from './actions';
+import { enhanceWithSiteType, recordTracksEvent } from 'state/analytics/actions';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { withEnhancers } from 'state/utils';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class UpworkStatsNudge extends Component {
+	static propTypes = {
+		isDismissed: PropTypes.bool.isRequired,
+		plan: PropTypes.object,
+		recordTracksEvent: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		siteSlug: PropTypes.string.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.recordView();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.siteId && this.props.siteId && this.props.siteId !== prevProps.siteId ) {
+			this.recordView();
+		}
+	}
+
+	recordView() {
+		if ( this.isVisible() ) {
+			this.props.recordTracksEvent( 'calypso_upwork_stats_nudge_view' );
+		}
+	}
+
+	onDismissClick = () => {
+		const plan = this.props.plan ? this.props.plan.productSlug : '';
+		this.props.recordTracksEvent( 'calypso_upworks_stats_nudge_dismiss_icon_click', {
+			plan,
+		} );
+		this.props.dismissNudge();
+	};
+
+	onStartNowClick = () => {
+		const plan = this.props.plan ? this.props.plan.productSlug : '';
+		this.props.recordTracksEvent( 'calypso_upwork_stats_nudge_start_now_button_click', {
+			plan,
+		} );
+	};
```

Comment:
Just a suggestion, but we could remove duplicate code with something like this:

```javascript
recordClick = ( eventName ) => {
  const plan = this.props.plan ? this.props.plan.productSlug : '';

  this.props.recordTracksEvent( eventName, { plan } );
};

onDismissClick = () => {
  this.recordClick( 'calypso_upworks_stats_nudge_dismiss_icon_click' );

  this.props.dismissNudge();
};

onStartNowClick = () => {
  this.recordClick( 'calypso_upwork_stats_nudge_start_now_button_click' );
};
```

Reply:
Perfect!